### PR TITLE
[pkg/otlp/model] Backport attributes changes from Datadog exporter

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?

Backports OpenTelemetry Collector Contrib Datadog exporter `attributes` package changes made in this [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6086) to `pkg/otlp/model`.

### Motivation

Keep `pkg/otlp/model` updated with latest changes to the `attributes` package in the OpenTelemetry Collector Contrib Datadog exporter. This is part of the effort led in the following [PR](https://github.com/DataDog/datadog-agent/pull/9162).

### Describe how to test your changes

The tests in the OpenTelemetry Collector Contrib Datadog exporter `attributes` package have also been backported.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
Looks good, but let's wait for the other PR to be merged before merging this one

**Comment by mackjmr** [comment]
@mx-psi the other PR was merged 👍

**Comment by mx-psi** [comment]
Let's merge then :)

**Pull request merged**
